README.md

```diff
@@ -114,21 +67,87 @@ where `bin/`, `lib/` etc. are. This should be the only argument that is
 mandatory, everything else should be optional. The default build type is
 Release, though this can be overridden.
 
-The following example shows how to specify the ComputeCpp location, then
-build and run the tests.
+The following command shows how to compile SYCL-DNN.
 
 ```bash
 # Setup build environment
 mkdir build && cd build
 cmake .. -DComputeCpp_DIR=/path/to/computecpp
 # Compile SYCL-DNN
 make -j`nproc`
+```
+
+### Sample Code
+
+The "samples" directory contains sample code for the 2D convolution and pooling 
+operations offered by SYCL-DNN. 
+
+These binaries are compiled when building SYCL-DNN using CMake.
+
+### Running the SYCL-DNN Tests
+
+The SYCL-DNN tests are compiled when building SYCL-DNN using CMake.
+The following command shows how to run the tests.
+
+```bash
 # Run the tests
 ctest
 # If compiled with benchmark support, run just the benchmarks
 ctest -C Benchmark -E test
 ```
 
+## Support
+
+### Bug reports and Issues
+
+Bug reports are vital to provide feedback to the developers about what is going
+wrong with the project, you can raise these using the ["Issues"](https://github.com/codeplaysoftware/SYCL-DNN/issues) 
+feature in GitHub.
+
+Please make sure that your bug report contains the following information:
+
+* A clear and descriptive title.
+
+* The output of
+  `clinfo | grep -E "Platform ID|Name|Vendor|[Vv]ersion|Profile|Extensions"`.
+
+* The output of `computecpp_info`.
+
+* The exact steps and commands to run to reproduce the bug.
+
+* The exact error text shown (if applicable), otherwise the behaviour you
+  expected and what you encountered instead.
+
+* Should the problem arise outside the project's test suite then please provide
+  a minimal test to allow us to reproduce the problem.
+
+## Project layout
+
+* bench/
+    - Some benchmarks, used to track the performance of SYCL-DNN.
+* cmake/
+    - Contains helper files and functions for downloading dependencies
+      and performing common tasks (like adding tests, libraries etc.).
+* CMakeLists.txt
+    - The root CMakeLists.txt file, refer to this when building SYCL-DNN.
+* CONTRIBUTING.md
+    - Information about how to contribute bug reports, code, or other works
+      to this project.
+* hooks/
+    - Contains scripts that are suitable for adding to the .git/hooks folder
+      (e.g. making sure that the repo is correctly formatted).
+* include/
+    - The directory under which the public interface is stored.
+* LICENSE
+    - The license this sofware is available under: Apache 2.0
+* README.md
+    - This readme file.
+* src/
+    - The source files of SYCL-DNN are kept in here.
+* test/
+    - All tests are bundled in this folder and are the mark of correctness
+      when doing pre-merge tests.
+
 ## Cross-compilation with ComputeCpp
```

Comment:
Why is this not near the `Building SYCL-DNN` section?

Reply:
It's quite a long section and doesn't seem like it would be relevant to everyone so I pushed it further down on its own. I can put it next to the build step but it then breaks up the flow if you don't want to cross compile